InnerEye/ML/metrics.py

```diff
@@ -242,14 +242,23 @@ def calculate_metrics_per_class(segmentation: np.ndarray,
                          f"the label tensor indicates that there are {number_of_classes - 1} classes.")
     binaries = binaries_from_multi_label_array(segmentation, number_of_classes)
 
-    all_classes_are_binary = [is_binary_array(ground_truth[label_id]) for label_id in range(ground_truth.shape[0])]
-    if not np.all(all_classes_are_binary):
+    binary_classes = [is_binary_array(ground_truth[label_id]) for label_id in range(ground_truth.shape[0])]
+
+    # If ground truth image is nan, then will not be used for metrics computation.
+    nan_images = [np.isnan(np.sum(ground_truth[label_id])) for label_id in range(ground_truth.shape[0])]
```

Comment:
np.isnan(np.sum(..)) is true if even a single entry in the ground truth ID is NaN. is that the check you wanted to implement?
Also, this is a very costly check to run. Have you considered an alternative, for example passing in a None entry in ground_truth_ids?

Reply:
Also, generally such special cases need to be documented clearly.